src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/CreateCohortController.cs

```diff
@@ -88,21 +98,57 @@ public async Task<IActionResult> SelectProvider(SelectProviderViewModel request)
                     ModelState.AddModelError(nameof(request.ProviderId), "Check UK Provider Reference Number");
                     return View(request);
                 }
-                _logger.LogError($"Failed '{nameof(CreateCohortController)}-{nameof(SelectProvider)}': {nameof(ex.StatusCode)}='{ex.StatusCode}', {nameof(ex.ReasonPhrase)}='{ex.ReasonPhrase}'");
+
+                _logger.LogError(
+                    $"Failed '{nameof(CreateCohortController)}-{nameof(SelectProvider)}': {nameof(ex.StatusCode)}='{ex.StatusCode}', {nameof(ex.ReasonPhrase)}='{ex.ReasonPhrase}'");
             }
             catch (Exception ex)
             {
-                _logger.LogError($"Failed '{nameof(CreateCohortController)}-{nameof(SelectProvider)}': {nameof(ex.Message)}='{ex.Message}', {nameof(ex.StackTrace)}='{ex.StackTrace}'");
+                _logger.LogError(
+                    $"Failed '{nameof(CreateCohortController)}-{nameof(SelectProvider)}': {nameof(ex.Message)}='{ex.Message}', {nameof(ex.StackTrace)}='{ex.StackTrace}'");
             }
 
-            return RedirectToAction("Error","Error");
+            return RedirectToAction("Error", "Error");
         }
 
         [Route("confirm-provider")]
         [HttpGet]
-        public IActionResult ConfirmProvider(ConfirmProviderRequest request)
+        public async Task<IActionResult> ConfirmProvider(ConfirmProviderRequest request)
+        {
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("Error", "Error");
+            }
+
+            var response = await _commitmentsApiClient.GetProvider(request.ProviderId);
+
+            var model = _confirmProviderViewModelMapper.Map(request);
+            model.ProviderId = response.ProviderId;
+            model.ProviderName = response.Name;
+
+            return View(model);
+        }
+
+        [Route("confirm-provider")]
+        [HttpPost]
+        public IActionResult ConfirmProvider(ConfirmProviderViewModel request)
         {
-            return View(request);
+            var validationResult = _confirmProviderViewModelValidator.Validate(request);
```

Comment:
I don't think this needs to called explicitly, it should occur automatically in the pipeline

Reply:
done